src/renderer/mod.rs

```diff
@@ -0,0 +1,244 @@
+//! The renderer for [`Snippet`]s
+//!
+//! # Example
+//! ```
+//! use annotate_snippets::renderer::Renderer;
+//! use annotate_snippets::snippet::{Annotation, AnnotationType, Slice, Snippet};
+//! let snippet = Snippet {
+//!     title: Some(Annotation {
+//!         label: Some("mismatched types"),
+//!         id: None,
+//!         annotation_type: AnnotationType::Error,
+//!     }),
+//!     footer: vec![],
+//!     slices: vec![
+//!         Slice {
+//!             source: "Foo",
+//!             line_start: 51,
+//!             origin: Some("src/format.rs"),
+//!             fold: false,
+//!             annotations: vec![],
+//!         },
+//!         Slice {
+//!             source: "Faa",
+//!             line_start: 129,
+//!             origin: Some("src/display.rs"),
+//!             fold: false,
+//!             annotations: vec![],
+//!         },
+//!     ],
+//!  };
+//!
+//!  let renderer = Renderer::styled();
+//!  println!("{}", renderer.render(snippet));
+
+pub mod margin;
+
+use crate::display_list::DisplayList;
+use crate::snippet::Snippet;
+use anstyle::{AnsiColor, Effects, Style};
```

Comment:
note: we likely want to re-export `anstyle`

Some people literally re-export the crate in the root (`pub use anstyle;`).  Where possible, I prefer doing it in a way that makes sense from an API perspective.  For example, in clap we have [mod styling](https://docs.rs/clap/latest/clap/builder/styling/index.html) which has both clap's Stylesheeting and all of `anstyle`